Dip/Dip/Definition.swift

```diff
@@ -50,20 +50,62 @@ func ==(lhs: DefinitionKey, rhs: DefinitionKey) -> Bool {
 public enum ComponentScope {
   /// Indicates that a new instance of the component will be created each time it's resolved.
   case Prototype
+  /// Indicates that instances will be reused during resolve but will be discurded when topmost `resolve` method returns.
+  case ObjectGraph
   /// Indicates that resolved component should be retained by container and always reused.
   case Singleton
 }
 
 ///Definition of type T describes how instances of this type should be created when they are resolved by container.
 public final class DefinitionOf<T>: Definition {
```

Comment:
But if that will be still method on a `struct` then it will mutate it or produce new instance instead of affecting one that is in registry. We could make it an optional argument on `register` method, but that will break using factory as trailing closure (don't know if you care about that)